rest/config_new.go

```diff
@@ -0,0 +1,240 @@
+
+	TLS  TLSConfig   `json:"tls,omitempty"`
+	CORS *CORSConfig `json:"cors,omitempty"`
+}
+
+type TLSConfig struct {
+	MinimumVersion string `json:"minimum_version,omitempty"     help:"The minimum allowable TLS version for the REST APIs"`
+	CertPath       string `json:"cert_path,omitempty" help:"The TLS cert file to use for the REST APIs"`
+	KeyPath        string `json:"key_path,omitempty"  help:"The TLS key file to use for the REST APIs"`
+}
+
+type CORSConfig struct {
+	Origin      []string `json:"origin,omitempty"       help:"List of allowed origins, use ['*'] to allow access from everywhere"`
+	LoginOrigin []string `json:"login_origin,omitempty" help:"List of allowed login origins"`
+	Headers     []string `json:"headers,omitempty"      help:"List of allowed headers"`
+	MaxAge      int      `json:"max_age,omitempty"      help:"Maximum age of the CORS Options request"`
+}
+
+type LoggingConfig struct {
+	LogFilePath    string                    `json:"log_file_path,omitempty"   help:"Absolute or relative path on the filesystem to the log file directory. A relative path is from the directory that contains the Sync Gateway executable file"`
+	RedactionLevel base.RedactionLevel       `json:"redaction_level,omitempty" help:"Redaction level to apply to log output"`
+	Console        *base.ConsoleLoggerConfig `json:"console,omitempty"`
+	Error          *base.FileLoggerConfig    `json:"error,omitempty"`
+	Warn           *base.FileLoggerConfig    `json:"warn,omitempty"`
+	Info           *base.FileLoggerConfig    `json:"info,omitempty"`
+	Debug          *base.FileLoggerConfig    `json:"debug,omitempty"`
+	Trace          *base.FileLoggerConfig    `json:"trace,omitempty"`
+	Stats          *base.FileLoggerConfig    `json:"stats,omitempty"`
+}
+
+type AuthConfig struct {
+	Facebook *FacebookConfig `json:"facebook,omitempty"`
+	Google   *GoogleConfig   `json:"google,omitempty"`
+}
+
+type FacebookConfig struct {
+	Register bool `json:"register,omitempty" help:"If true, server will register new user accounts"`
+}
+
+type GoogleConfig struct {
+	Register    bool     `json:"register,omitempty" help:"If true, server will register new user accounts"`
+	AppClientID []string `json:"app_client_id,omitempty" help:"List of enabled client ids"`
+}
+
+type ReplicatorConfig struct {
+	MaxHeartbeat    time.Duration `json:"max_heartbeat,omitempty"    help:"Max heartbeat value for _changes request"`
+	BLIPCompression *int          `json:"blip_compression,omitempty" help:"BLIP data compression level (0-9)"`
+}
+
+type UnsupportedConfig struct {
+	StatsLogFrequency *time.Duration `json:"stats_log_frequency,omitempty" help:"How often should stats be written to stats logs"`
+	UseStdlibJSON     bool           `json:"use_stdlib_json,omitempty"     help:"Bypass the jsoniter package and use Go's stdlib instead"`
+
+	HTTP2 *HTTP2Config `json:"http2,omitempty"`
+}
+
+type HTTP2Config struct {
```

Comment:
Having Http2Config and HTTP2Config in the same package seems error-prone.  Can we remove the old one, since only the annotations differ?